### PR TITLE
TIMEOUT 0 set timeout to INT32_MAX

### PR DESCRIPTION
**Pull request by ashtul**
fixes issue #1987

### CODE REVIEW

File: tests/pytests/test_config.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def testAllConfig(env):
     env.assertEqual(res_dict['MAXAGGREGATERESULTS'][0], 'unlimited')
     env.assertEqual(res_dict['MAXEXPANSIONS'][0], '200')
     env.assertEqual(res_dict['MAXPREFIXEXPANSIONS'][0], '200')
-    env.assertIn(res_dict['TIMEOUT'][0], ['500', '100000'])
+    env.assertIn(res_dict['TIMEOUT'][0], ['500', '2147483647'])
```
Comment (MeirShpilraien) [inline tests/pytests/test_config.py:97]: Not sure I follow those test changes... why was those changes needed?

File: tests/pytests/CMakeLists.txt [hunk 2]
```diff
@@ -12,7 +12,7 @@ IF (RS_VERBOSE_TESTS)
 ENDIF()
 
 FILE(GLOB PY_TEST_FILES "test*.py")
-SET(baseCommand "MODARGS+='timeout 100000;' ${CMAKE_CURRENT_SOURCE_DIR}/runtests.sh --module=${RS_TEST_MODULE_SO} ${RS_TEST_ARGS}")
+SET(baseCommand "MODARGS+='timeout 0;' ${CMAKE_CURRENT_SOURCE_DIR}/runtests.sh --module=${RS_TEST_MODULE_SO} ${RS_TEST_ARGS}")
```
Comment (MeirShpilraien) [inline tests/pytests/CMakeLists.txt:15]: What is this change means?

**Review by MeirShpilraien — COMMENTED** [review]
:+1: 
2 small questions...

File: tests/pytests/CMakeLists.txt (hunk 2 above)
Comment (ashtul) [reply tests/pytests/CMakeLists.txt:15]: I have changed the default of `100000` to `0` using the new feature.
We had to put the default because some tests have timed out on the sanitizer tests.

File: tests/pytests/test_config.py (hunk 1 above)
Comment (ashtul) [reply tests/pytests/test_config.py:97]: This change reflects the change in `tests/pytests/CMakeLists.txt`.

Comment (MeirShpilraien) [reply tests/pytests/test_config.py:97]: So on config get it means that we will not get 0 but '2147483647'? Isn't it confusing?

Comment (ashtul) [reply tests/pytests/test_config.py:97]: We can print `disabled` instead. WDYT?

Comment (MeirShpilraien) [reply tests/pytests/test_config.py:97]: Sounds good, but why are we getting this number and not zero?

Comment (ashtul) [reply tests/pytests/test_config.py:97]: Look at config.c 196-198.

Comment (MeirShpilraien) [reply tests/pytests/test_config.py:97]: @ashtul why not return 0 ?

Comment (ashtul) [reply tests/pytests/test_config.py]: @oshadmi Good idea. Implemented!

**Review by oshadmi — APPROVED** [review]
👍🏼
